src/lib/node/json_dao.js

```diff
@@ -20,6 +20,8 @@ foam.CLASS({
   name: 'JSONFileDAO',
   extends: 'foam.dao.ArrayDAO',
```

Comment:
Why ArrayDAO instead of an MDAO? With ArrayDAO, find() and remove() are going to be O(n) without a cache, and remove() is going to be O(n) even with a cache.